### PR TITLE
Cairo1-run compatibility

### PR DESCRIPTION
**Pull request by m-kus**
This PR:
1. Introduces new scarb profile `cairo1-run` which configures cairo compiler in a special way
2. Sets integration testing entrypoint as main (we don't use main anyways) because cairo1-run cannot run arbitrary functions
3. Updates scripts accordingly
4. Misc: makes gas metering more accurate

How to use:
```
cargo install --git https://github.com/lambdaclass/cairo-vm cairo1-run
scarb --profile cairo1-run build
python scripts/data/format_args.py --input_file packages/client/tests/data/light_169.json --cairo1_run > args.json
cairo1-run --args_file args.json --layout all_cairo target/cairo1-run/client.sierra.json
```

### CODE REVIEW

File: Scarb.toml [hunk 1]
```diff
@@ -11,5 +11,10 @@ license-file = "LICENSE"
 
 [workspace.dependencies]
 cairo_test = "2.8.0"
-
 shinigami_engine = { git = "https://github.com/keep-starknet-strange/shinigami.git", rev = "9e7692d" }
+
+[profile.cairo1-run.cairo]
+# https://github.com/lambdaclass/cairo-vm/issues/1745
+enable-gas = false
```
Comment (maciejka) [inline Scarb.toml:18]: Out of curiosity, what does this disable?

File: scripts/data/format_args.py [hunk 2]
```diff
@@ -110,14 +110,18 @@ def format_item(item):
...
     """Reads arguments from JSON file and prints formatted result.
     Expects a single CLI argument containing file path.
     Output is compatible with the Scarb runner arguments format.
     """
     args = json.loads(Path(input_file).read_text())
     res = flatten_tuples(serialize(args))
-    print([res, 1 if execute_script else 0])
+    flag = 1 if execute_script else 0
+    if cairo1_run:
+        print(f"{format_cairo1_run(res)} {flag}")
```
Comment (maciejka) [inline scripts/data/format_args.py:122]: What is `format_cairo1_run`?

File: Scarb.toml (hunk 1 above)
Comment (m-kus) [reply Scarb.toml:18]: I think gas related instructions, but scarb cairo-run doesn't work without `enable-gas = true` even if you remove `get_gas_available` function usage

File: scripts/data/format_args.py (hunk 2 above)
Comment (m-kus) [reply scripts/data/format_args.py:122]: cairo1-run uses a different arguments format vs scarb
format_cairo1_run is a function that was always there, just not used :)

**Review by maciejka — APPROVED** [review]

**Pull request merged**